tests/hilti/validation/duplicate.hlt

```diff
@@ -63,10 +68,33 @@ function uint<8> fun(uint<8> a) {
     return uint8(1);
 }
 
-function uint<16> fun(uint<8> a, uint<8> b = uint8(0)) {
-    return uint16(2);
+function uint<8> fun(uint<8> a, uint<8> b = uint8(0)) {
+    return uint8(2);
 }
 
 hilti::print(fun(uint8(1)));
 
+function uint<8> diff_return(uint<8> a) {
+    return uint8(1);
+}
+
+function uint<16> diff_return(uint<8> a, uint<8> b = uint8(0)) {
+    return uint16(2);
+}
+
+hilti::print(diff_return(uint8(1)));
+
+}
+
+# @TEST-START-NEXT
+
+module Foo {
+
+# Equivalent functions signatures should not be allowed.
+function void fail1() { return; }
+function void fail1() { return; }
+
+function string fail2(uint<8> x, string y) { return "hi!"; }
+function string fail2(uint<8> different, string names) { return "hi!"; }
+
```

Comment:
Could you add a variant where the signatures are identical, but the bodies differ? We do not look at bodies when checking for overloads, so we'd expect them to be rejected as well, e.g.,

```suggestion
function uint<8> fail3() { return 1; }
function uint<8> fail3() { return 2; }
```